### PR TITLE
DDF-2083 Fix password values within configurations to not be visible when inspected via chrome/developer console.

### PR DESCRIPTION
**Pull request by roelens8**
#### What does this PR do?
Passwords values within any configuration will not be visible when inspecting its HTML input element via the chrome/developer console.
#### Who is reviewing it (please choose AT LEAST two reviewers that need to approve the PR before it can get merged; if a component team is listed, at least one of its members needs to approve)?
@ryeats @coyotesqrl @tbatie @bcwaters @jckilmer 
#### Choose 2 committers to review/merge the PR (please choose ONLY two committers from below, delete the rest).
@pklinef
@stustison
#### How should this be tested?
-Verify that when changing a password to its correct value within some configuration, the password value is shown as "password" when inspecting its input element via the chrome/developer console. Then verify the password is still correct by testing the feature containing that password property, still works. 
#### What are the relevant tickets?
DDF-2083
#### Checklist:
- [ ] Documentation Updated
- [X] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

### CODE REVIEW

File: catalog/admin/admin-poller-service/src/main/java/org/codice/ddf/catalog/admin/poller/AdminPollerServiceBean.java [hunk 1]
```diff
@@ -229,10 +230,28 @@ public boolean sourceStatus(String servicePID) {
...
+                            // If the configuration property is a password, set its value to
+                            // "password" if it's not empty, so that the real password value will
+                            // be hidden. (Passwords within source configs are usually optional,
+                            // thus it should have the ability to update to an empty string).
+                            for (Map<String, Object> metatypeProperty : metatypeProperties) {
+                                if (metatypeProperty.get("id")
+                                        .equals(key) &&
+                                        AttributeDefinition.PASSWORD
+                                                == (Integer) metatypeProperty.get("type")
+                                        && !"".equals(propertyValue)) {
```
Comment (coyotesqrl) [inline catalog/admin/admin-poller-service/src/main/java/org/codice/ddf/catalog/admin/poller/AdminPollerServiceBean.java]: You're looping unnecessarily over anything that has a `propertyValue` of an empty string. This whole nested loop can be rewritten much more efficiently as something like...

```java
                        while (configProperties.keys()
                                .hasMoreElements()) {
                            String key = configProperties.keys()
                                    .nextElement();
                            propertyValue = configProperties.get(key);
                            if (!propertyValue.equals("")) {
                                for (Map<String, Object> metatypeProperty : metatypeProperties) {
                                    if (metatypeProperty.get("id")
                                            .equals(key) && AttributeDefinition.PASSWORD
                                            == (Integer) metatypeProperty.get("type")) {
                                        propertyValue = "password";
                                        break;
                                    }
                                }
                            }
                            plist.put(key, propertyValue);
                        }
```

File: platform/admin/core/admin-core-api/src/main/java/org/codice/ddf/ui/admin/api/ConfigurationAdminExt.java [hunk 2]
```diff
@@ -316,6 +315,18 @@ private void addConfigurationData(Map<String, Object> service, Configuration[] c
...
             }
 
+            // If the configuration property is a password, set its value to "password" so that
+            // the real password value will be hidden.
+            List<HashMap<String, Object>> metatypeList =
+                    (List<HashMap<String, Object>>) service.get("metatype");
+            metatypeList.stream()
+                    .filter(metatype -> AttributeDefinition.PASSWORD == (Integer) metatype.get(
+                            "type"))
+                    .forEach(metatype -> {
```
Comment (coyotesqrl) [inline platform/admin/core/admin-core-api/src/main/java/org/codice/ddf/ui/admin/api/ConfigurationAdminExt.java]: I don't generally like side-effects like this, but it probably cleaner and clearer than most of the other options. You could consider something like a map to get the `passwordProperty` and then a collect to a map, then at the end call `putAll` to your existing `propertiesTable`, but that's probably a bit less efficient.

File: catalog/admin/admin-poller-service/src/main/java/org/codice/ddf/catalog/admin/poller/AdminPollerServiceBean.java (hunk 1 above)
Comment (roelens8) [inline catalog/admin/admin-poller-service/src/main/java/org/codice/ddf/catalog/admin/poller/AdminPollerServiceBean.java]: Fixed the unnecessary looping over the empty string.

File: platform/admin/core/admin-core-api/src/main/java/org/codice/ddf/ui/admin/api/ConfigurationAdminExt.java (hunk 2 above)
Comment (roelens8) [inline platform/admin/core/admin-core-api/src/main/java/org/codice/ddf/ui/admin/api/ConfigurationAdminExt.java]: Thanks for the suggestion, but I agree this way is cleaner and clearer, so I will keep it that way,

**Comment by tbatie** [comment]
:+1: Followed heroing steps successfully

**Comment by pklinef** [comment]
👍

**Pull request merged**